src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -184,17 +184,24 @@
 (defn build-whitelist-authorizer
   "Build a function that will authorize requests based on the supplied
   certificate whitelist (see `cn-whitelist->authorizer` for more
-  details). Rejected requests are logged."
+  details). Returns :authorized if the request is allowed, otherwise a
+  string describing the reason not."
   [whitelist]
   {:pre  [(string? whitelist)]
    :post [(fn? %)]}
   (let [allowed? (kitchensink/cn-whitelist->authorizer whitelist)]
     (fn [{:keys [ssl-client-cn] :as req}]
       (if (allowed? req)
-        true
+        :authorized
         (do
           (log/warnf "%s rejected by certificate whitelist %s" ssl-client-cn whitelist)
-          false)))))
+          (format (str "The client certificate name (%s) doesn't "
+                       "appear on the certificate whitelist. Is your "
+                       "master's certname listed in PuppetDB's "
+                       "cetificate-whitelist file? If you recently "
```

Comment:
@pizzaops can you remind me again, why would a user get this far, if there certs are bad? What I mean is, why would puppetdb ssl-setup help them in this scenario? Surely if there certs were bad, they would get a completely different error before this one was ever hit, more to do with bad CA, or the pdb cert is not signed by a trusted CA etc. etc.